### PR TITLE
functions-and-operators: Improve format, add vitess_hash

### PR DESCRIPTION
**Pull request by TomShawn**
### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v6.6 (TiDB 6.6 versions)
- [x] v6.5 (TiDB 6.5 versions)
- [x] v6.4 (TiDB 6.4 versions)
- [x] v6.1 (TiDB 6.1 versions)
- [x] v5.4 (TiDB 5.4 versions)
- [x] v5.3 (TiDB 5.3 versions)
- [x] v5.2 (TiDB 5.2 versions)
- [x] v5.1 (TiDB 5.1 versions)
- [x] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs/pull/6558 https://github.com/pingcap/docs/pull/7468
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: functions-and-operators/tidb-functions.md [hunk 1]
```diff
@@ -256,13 +177,9 @@ Check Table Before Drop: false
...
 ```
 
-### MySQL 兼容性
-
-`TIDB_VERSION` 是 TiDB 特有的函数，和 MySQL 不兼容。如果要求兼容 MySQL，可以使用 `VERSION` 获取版本信息，但结果不包含详细的构建信息。
-
 ## TIDB_DECODE_SQL_DIGESTS
 
-`TIDB_DECODE_SQL_DIGESTS` 函数用于在集群中查询一组 SQL Digest 所对应的 SQL 语句的归一化形式（即去除格式和参数后的形式）。函数接受 1 个或 2 个参数：
+`TIDB_DECODE_SQL_DIGESTS（）` 函数用于在集群中查询一组 SQL Digest 所对应的 SQL 语句的归一化形式（即去除格式和参数后的形式）。函数接受 1 个或 2 个参数：
```
Comment (Oreoxmt) [inline functions-and-operators/tidb-functions.md:182]:
```suggestion
`TIDB_DECODE_SQL_DIGESTS()` 函数用于在集群中查询一组 SQL Digest 所对应的 SQL 语句的归一化形式（即去除格式和参数后的形式）。函数接受 1 个或 2 个参数：
```

File: functions-and-operators/date-and-time-functions.md [hunk 2]
```diff
@@ -7,6 +7,11 @@ aliases: ['/docs-cn/dev/functions-and-operators/date-and-time-functions/','/docs
 
 TiDB 支持使用 MySQL 5.7 中提供的所有[日期和时间函数](https://dev.mysql.com/doc/refman/5.7/en/date-and-time-functions.html)。
 
+> **注意：**
+>
+> - MySQL 常常会接受格式不正确的日期和时间值。例如，`'2020-01-01\n\t01:01:01'` 和 `'2020-01_01\n\t01:01'` 被视为有效的日期和时间值。
+> - TiDB 会尽量匹配 MySQL 的行为，但可能不会在所有情况下都匹配。建议使用正确的格式化日期，TiDB 文档中未记录将如何处理格式不正确的值。
```
Comment (Oreoxmt) [inline functions-and-operators/date-and-time-functions.md:13]:
```suggestion
> - TiDB 会尽量与 MySQL 的行为保持一致，但可能无法在所有情况下完全匹配。建议使用正确的格式化日期，TiDB 文档中未记录将如何处理格式不正确的值。
```

File: functions-and-operators/date-and-time-functions.md [hunk 3]
```diff
@@ -72,6 +77,25 @@ TiDB 支持使用 MySQL 5.7 中提供的所有[日期和时间函数](https://de
...
+TiDB 支持 `str_to_date()` 函数，但是无法解析所有的日期和时间值。此外，TiDB 不支持以下日期和时间格式化选项：
+
+| 格式 | 说明                                                                           |
+|--------|---------------------------------------------------------------------------------------|
+| "%a"   | 星期名的缩写（例如 Sun..Sat）                                                  |
+| "%D"   | 带英文后缀的月份日期（例如 0th，1st，2nd，3rd）                             |
+| "%U"   | 星期 (00..53)，星期日是每周的第一天；WEEK() mode 0               |
+| "%u"   | 星期 (00..53)，星期一是每周的第一天；WEEK() mode 1               |
+| "%V"   | 星期 (01..53)，星期日是每周的第一天；WEEK() mode 2；和 %X 一起使用 |
+| "%v"   | 星期 (01..53)，星期一是每周的第一天；WEEK() mode 3；和 %x 一起使用 |
```
Comment (Oreoxmt) [inline functions-and-operators/date-and-time-functions.md:91]:
```suggestion
| "%V"   | 星期 (01..53)，星期日是每周的第一天；WEEK() mode 2；和 "%X" 一起使用 |
| "%v"   | 星期 (01..53)，星期一是每周的第一天；WEEK() mode 3；和 "%x" 一起使用 |
```

File: functions-and-operators/tidb-functions.md [hunk 4]
```diff
@@ -5,7 +5,23 @@ summary: 学习使用 TiDB 特有的函数。
 
 # TiDB 特有的函数
 
-本文档介绍 TiDB 特有的函数。
+以下函数为 TiDB 中特有的函数，与 MySQL 不兼容：
+
+| 函数名 | 函数说明 |
+| :-------------- | :------------------------------------- |
+| [`TIDB_BOUNDED_STALENESS()`](#tidb_bounded_staleness) |  `TIDB_BOUNDED_STALENESS` 函数指示 TiDB 在时间范围内读取尽可能新的数据。参见[使用 AS OF TIMESTAMP 语法读取历史数据](/as-of-timestamp.md)。 |
```
Comment (Oreoxmt) [inline functions-and-operators/tidb-functions.md:12]:
```suggestion
| [`TIDB_BOUNDED_STALENESS()`](#tidb_bounded_staleness) |  `TIDB_BOUNDED_STALENESS` 函数指示 TiDB 在指定时间范围内读取尽可能新的数据。参见[使用 AS OF TIMESTAMP 语法读取历史数据](/as-of-timestamp.md)。 |
```

**Review by Oreoxmt — APPROVED** [review]
LGTM

**Comment by qiancai** [comment]
Removed the **needs-cherry-pick-release-6.3** label because the v6.3 docs have been archived at https://docs-archive.pingcap.com/zh/tidb/v6.3 and will no longer receive new updates.

**Comment by qiancai** [comment]
Removed the needs-cherry-pick-release-6.4 label because the v6.4 docs have been archived at https://docs-archive.pingcap.com/zh/tidb/v6.4 and will no longer receive new updates.

**Review by ran-huang — APPROVED** [review]

File: functions-and-operators/tidb-functions.md [hunk 5]
```diff
@@ -5,27 +5,29 @@ summary: 学习使用 TiDB 特有的函数。
 
 # TiDB 特有的函数
 
-本文档介绍 TiDB 特有的函数。
+以下函数为 TiDB 中特有的函数，与 MySQL 不兼容：
 
-## TIDB_BOUNDED_STALENESS
+| 函数名 | 函数说明 |
+| :-------------- | :------------------------------------- |
+| [`TIDB_BOUNDED_STALENESS()`](#tidb_bounded_staleness) |  `TIDB_BOUNDED_STALENESS` 函数指示 TiDB 在指定时间范围内读取尽可能新的数据。参见[使用 AS OF TIMESTAMP 语法读取历史数据](/as-of-timestamp.md)。 |
```
Comment (Oreoxmt) [inline functions-and-operators/tidb-functions.md:12]:
```suggestion
| `TIDB_BOUNDED_STALENESS()` |  `TIDB_BOUNDED_STALENESS` 函数指示 TiDB 在指定时间范围内读取尽可能新的数据。参见[使用 AS OF TIMESTAMP 语法读取历史数据](/as-of-timestamp.md)。 |
```

**Pull request merged**